libeventheader-decode-cpp/src/EventFormatter.cpp

```diff
@@ -782,48 +882,206 @@ WriteUcsVal(
     assert(9u <= sb.Room());
     if (json)
     {
-        sb.WriteUtf8Byte('"');
+        sb.WriteUtf8ByteUnchecked('"');
         sb.WriteUcsCharJsonEscaped(ucs4); // UCS1
-        sb.WriteUtf8Byte('"');
+        sb.WriteUtf8ByteUnchecked('"');
     }
     else
     {
         sb.WriteUcsChar(ucs4); // UCS1
     }
 }
 
+// Requires: there is room for 18 chars.
+static void
+WriteFloat32(
```

Comment:
Note: Now that we're supporting nullable fields, the WriteFloat code is needed in two places, so factor-out a subroutine for it.